veracruz-client/src/veracruz_client.rs

```diff
@@ -172,21 +171,32 @@ impl VeracruzClient {
         client_cert_filename: P,
         public_key: &mut Pk,
     ) -> Result<()> {
-        let cert_file = File::open(&client_cert_filename)?;
-        let parsed_cert = x509_parser::pem::Pem::read(BufReader::new(cert_file))?;
-        let parsed_cert = parsed_cert.0.parse_x509()?.tbs_certificate;
-        let cert_public_key_der =
-            Pk::from_public_key(parsed_cert.subject_pki.subject_public_key.data)?
-                .write_public_der_vec()?;
+        // Read and parse certificate.
+        let mut buffer = VeracruzClient::read_all_bytes_in_file(&client_cert_filename)?;
+        buffer.push(b'\0');
+        let mut cert = Certificate::from_pem(&buffer)?;
 
+        // Compare public keys as DER.
+        let cert_public_key_der = cert.public_key_mut().write_public_der_vec()?;
         let public_key_der = public_key.write_public_der_vec()?;
         if cert_public_key_der != public_key_der {
-            Err(anyhow!(VeracruzClientError::UnexpectedKey))
-        } else if parsed_cert.validity.time_to_expiration().is_none() {
-            Err(anyhow!(VeracruzClientError::UnexpectedCertificate))
-        } else {
-            Ok(())
+            return Err(anyhow!(VeracruzClientError::UnexpectedKey));
+        }
+
+        // Check validity period.
+        #[cfg(features = "std")]
```

Comment:
Do we need "std" checks in the Client?

Reply:
I found we did, though I didn't investigate why.

Reply:
I would have thought it's best to leave it in for now. We'll probably never need the client under IceCap, but it's working now, so why break it?

Reply:
Good point